src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -18,6 +19,9 @@ internal partial class Checksum
         private static readonly ObjectPool<IncrementalHash> s_incrementalHashPool =
             new ObjectPool<IncrementalHash>(() => IncrementalHash.CreateHash(HashAlgorithmName.SHA1), size: 20);
 
+        public static Checksum Create(string val)
```

Comment:
can we not use MemoryStream and create new byte array for string?

Reply:
Yeah.  let me see if i can find a better way to do this that produces less garbage.

Reply:
Moved to a non-allocating version.